File: API/Entities/CustomerProductList.cs
namespace API.Entities
{
    public class CustomerProductList
    {
        public int Id { get; set; }
        public string RowGuid { get; set; }
        public int ProductId { get; set; }
        public string Version { get; set; }
        public DateTime FirstDate { get; set; }
        public DateTime EndDate { get; set; }
        public int CustomerId { get; set; }
        public string Description { get; set; }
    }
}

File: Client/src/app/_models/customerProductList.ts
export interface CustomerProductList {
    id: number;
    rowGuiid: string;
    productId: number;
    version: string;
    firstDate: string;
    endDate: string;
    customerId: number;
    customerName: string;
    description: string;
    productName: string;
}

export interface CustomerProductListAdd {
    productId: number;
    version: string;
    firstDate: string;
    endDate: string;
    customerId: number;
    description: string;
    productName: string;
}